### PR TITLE
MacOS support

### LINKED ISSUE #2: Support macOS

### PR DESCRIPTION
**Pull request by Salzian**
The main challenge was to understand how rust uses the MacOS linker. There is barely any documentation online that is helpful to this specific use case. It took lots of trial and error to get to a point, where a crate depending on `bevy_fmod` can simply use the dylibs when using `cargo run`. 

Using packaging tooling like `cargo-bundle` requires some further configuration of the linking paths, but that's out of scope for this crate.

Closes #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,20 +35,41 @@ This requires a free FMOD account.
 
 ### Windows
 
-- Download the "FMOD Engine" package for Windows.
+- Download the "FMOD Studio" and ""FMOD Engine" package for Windows.
```
Comment (GitGhillie) [inline README.md:38]: The FMOD Engine package contains both the core and the studio libs, so I'm not sure why this change is necessary

File: README.md [hunk 2]
```diff
@@ -35,20 +35,41 @@ This requires a free FMOD account.
...
     - `api/studio/lib/x64/fmodstudio_vc.lib`
 
+### MacOS
+
+- Download the "FMOD Studio" and "FMOD Engine" package for MacOS.
+- Move these files:
+    - `api/core/lib/libfmod.dylib`
+    - `api/studio/lib/libfmodL.dylib` (debugging only)
+    - `api/studio/lib/libfmodstudio.dylib`
+    - `api/studio/lib/libfmodstudioL.dylib` (debugging only)
```
Comment (GitGhillie) [inline README.md:53]: I don't think we've mentioned the debugging (or Logging) version of the libs before, should we explain how they are used? (Could be a follow-up PR)

**Review by GitGhillie — COMMENTED** [review]
I think it would make sense to include the libraries in a separate PR and then include the libraries for all the platforms in one go. Or do the macos libs have a bigger reason to be included with the repo?

File: README.md (hunk 1 above)
Comment (Salzian) [reply README.md:38]: original brainfart moment. I will have another look and write the definitive guide.

File: README.md (hunk 2 above)
Comment (Salzian) [reply README.md:53]: I have seen that libfmod automatically loads one or the other depending on some cargo emv var. I will further investigate in a follow-up and remove the mention here for now.

**Comment by msvbg** [comment]
Hey, nice work! I was just about to have a look at this, but then it turns out you already did the work. I tried this PR in my game and it seems to work as intended 🎉

**Comment by Salzian** [comment]
@msvbg Nice, thanks for testing it.

**Comment by msvbg** [comment]
One little nit about this approach is that `cargo test --workspace` seems to not like the relative paths in `rustflags`, probably because the test program gets invoked from a directory that isn't the workspace root. There should be some sort of fix for this but haven't found it yet.

**Comment by Salzian** [comment]
@msvbg Good catch. My testing was done in non-workspace projects so far. This is definitely a case that needs to be addressed somehow. Maybe we do need a build script after all where the user provides the paths of the libraries.

**Comment by msvbg** [comment]
Fwiw I am still using this branch and it works well for me. Workspace support would be nice to have but is not a dealbreaker, especially since you can just make your game a root-level cargo package alongside your workspace which side-steps these issues.

**Comment by Salzian** [comment]
@msvbg I'm working on this PR every now and then when I have some time. Workspace support is obviously a requirement as I use workspaces a lot myself. Glad this branch / approach works for you right now until I merge a final version. Thank you for your patience.

**Comment by Salzian** [comment]
@msvbg My current learning on this matter is, that I believe everyone's setup is different. The `.cargo/config.toml` of this project should not interfere with your setup. It shall only be needed for building `bevy_fmod` from source. I edited the `README.md` extensively to provide more information on linking dynamic libraries in Rust. In the end, everyone should write their own config tailored to their needs.

File: README.md [hunk 3]
```diff
@@ -1,54 +1,141 @@
...
+
+The second line will determine the rpath of the executable. This is the path
+where the executable will look for the libraries. By default, executables will
+look in a variety of places, including the directory the executable is in. This
+is fine when publishing the game, as you can just use the Windows method and put
+the libraries in the same directory as the executable. However, during
+development, the executable is in the `target/debug` directory, which gets
+generated automatically by cargo and does not contain the libraries. The
+`"-C", "link-arg=-Wl,-rpath,./vendor/macos"` flag will tell the executable to
+look in the `vendor/fmod` directory of your project for the libraries.
```
Comment (GitGhillie) [inline README.md:107]: The path mentioned is relative to the project root, which the executable doesn't know about. So I assume you mean to say that it will tell the executable to look there when using `cargo run`. Might be good to mention that (not sure how to do it succinctly though :cry: ).

File: README.md [hunk 4]
```diff
@@ -1,54 +1,141 @@
...
+
+By the end, your project structure should look like this:
```
Comment (GitGhillie) [inline README.md:109]: The Windows section contains instructions on what to do when publishing the game, which I'm not seeing here

**Review by GitGhillie — COMMENTED** [review]
I don't have a mac to test this but overall I'm feeling good about this. Might be worth creating a readme for every platform and linking to that in the main readme, so we can go in full detail for each platform

File: README.md [hunk 5]
```diff
@@ -1,54 +1,141 @@
...
+of the libraries in the same directory as the executable.
+
+```
+My Game/
+├── My Game.exe
+├── fmod.dll
+└── fmodstudio.dll
+```
+
+### MacOS
```
[the remark this answers is not in the document]
Comment (Salzian) [reply README.md:61]: On the other hand, titles are always capitablized in english, regardless of the word :shrug:

File: README.md (hunk 3 above)
Comment (Salzian) [reply README.md:107]: This information is for development only. When the user runs the program from an IDE, the IDE will set the working directory to the folder of the project and then execute the binary. That way the executable will be able to resolve the library path.

File: README.md (hunk 4 above)
Comment (Salzian) [reply README.md:109]: It's partially included in the previous paragraph. I really don't want to spent any more time writing docs. The linker settings depend highly on the user's setup and I guess this info here should be enough to figure out the last 5%.
